.github/workflows/asv-benchmarking.yml

```diff
@@ -0,0 +1,86 @@
+name: ASV Benchmarking
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    env:
+      CONDA_ENV_FILE: ./ci/asv.yml
+      ASV_DIR: ./benchmarks
+
+    steps:
+      - name: Checkout uxarray
+        uses: actions/checkout@v4
+        with:
+          repository: UXARRAY/uxarray
+          fetch-depth: 0
+      - name: Checkout uxarray-asv
+        uses: actions/checkout@v4
+        with:
+          repository: uxarray/uxarray-asv
+          persist-credentials: false
+          fetch-depth: 0
+          ref: main
+          path: uxarray-asv
+      - name: Set environment variables
+        run: |
+          echo "TODAY=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+
+      - name: Set up conda environment
+        id: env-setup
+        continue-on-error: true
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-file: ./ci/asv.yml
+          cache-environment: true
+          cache-environment-key: "benchmark-${{runner.os}}-${{runner.arch}}-${{env.TODAY}}"
+
+      - name: retry environment set up if failed
+        if: steps.env-setup.outcome == 'failure'
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          download-micromamba: false
+          environment-file: ./ci/asv.yml
+          cache-environment: true
+          cache-environment-key: "benchmark-${{runner.os}}-${{runner.arch}}-${{env.TODAY}}"
+
+      - name: Copy existing results
+        run: |
+          if [ -d "uxarray-asv/results" ]; then
+            cp -r uxarray-asv/results benchmarks/
+          fi
+
+      - name: Run benchmarks
+        shell: bash -l {0}
+        id: benchmark
+        run: |
+          cd benchmarks
+          asv machine --machine GH-Actions --os ubuntu-latest --arch x64 --cpu "2-core unknown" --ram 7GB
+          asv run v2024.02.0..main --skip-existing --parallel || true
+
+      - name: Commit and push benchmark results
+        run: |
+          if [ -d "uxarray-asv/results" ]; then
+            rm -r uxarray-asv/results
+          fi
+          cp -r benchmarks/results/ uxarray-asv/
+          cd uxarray-asv
+          git config --local user.email "chmielowiec.philip@gmail.com"
+          git config --local user.name "philipc2"
```

Comment:
Curious if this could be handled in a way independent of individual username/email

Reply:
I'll look into it! There should be a way to handle this using the UXarray organization.

Reply:
From what I've read, we still need to have a user & email of someone that is authorized to the repository in order to commit.

However, I've been able to successfully use a token from the UXarray organization for pushing, instead of a person one linked to my account. 

I think this should be fine?